sdk/communication/Azure.Communication.Common/src/MicrosoftTeamsAppIdentifier.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Communication
+{
+    /// <summary>Represents a Microsoft Teams app.</summary>
+    public class MicrosoftTeamsAppIdentifier : CommunicationIdentifier
+    {
+        private string _rawId;
+
+        /// <summary>
+        /// Returns the canonical string representation of the <see cref="MicrosoftTeamsAppIdentifier"/>.
+        /// You can use the <see cref="RawId"/> for encoding the identifier and then use it as a key in a database.
+        /// </summary>
+        public override string RawId
+        {
+            get
+            {
+                if (_rawId != null)
+                    return _rawId;
+
+                if (Cloud == CommunicationCloudEnvironment.Dod)
+                {
+                    _rawId = $"{TeamsAppDodCloud}{TeamsAppId}";
+                }
+                else if (Cloud == CommunicationCloudEnvironment.Gcch)
+                {
+                    _rawId = $"{TeamsAppGcchCloud}{TeamsAppId}";
+                }
+                else
+                {
+                    _rawId = $"{TeamsAppPublicCloud}{TeamsAppId}";
+                }
+
+                return _rawId;
+            }
+        }
+
+        /// <summary>The id of the Microsoft Teams app.</summary>
+        public string TeamsAppId { get; }
+
+        /// <summary> The cloud that the app belongs to. </summary>
+        public CommunicationCloudEnvironment Cloud { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MicrosoftTeamsAppIdentifier"/>.
+        /// </summary>
+        /// <param name="teamsAppId">Id of the Microsoft teams app. The unique Microsoft app ID for Teams.</param>
```

Comment:
```suggestion
        /// <param name="teamsAppId">The unique ID of the Microsoft Teams Application.</param>
```
maybe we need to update everywhere "app" to "Application", wdyt?

Reply:
Renamed.